src/parser/parser.ts

```diff
@@ -2168,30 +2184,58 @@ function parseBinaryExpression(
       );
     }
 
-    if (parser.linterFlags & LinterFlags.NoBitwise) {
-      switch (parser.token) {
-        case SyntaxKind.BitwiseAnd:
-        case SyntaxKind.BitwiseAndAssign:
-        case SyntaxKind.BitwiseOr:
-        case SyntaxKind.BitwiseOrAssign:
-        case SyntaxKind.BitwiseXor:
-        case SyntaxKind.BitwiseXorAssign:
-        case SyntaxKind.ShiftLeft:
-        case SyntaxKind.ShiftLeftAssign:
-        case SyntaxKind.ShiftRight:
-        case SyntaxKind.ShiftRightAssign:
-        case SyntaxKind.LogicalShiftRight:
-        case SyntaxKind.LogicalShiftRightAssign:
+    if (parser.linterFlags & LinterFlags.No) {
+      if (parser.subRules && SubRules.DoubleEquals && left.kind !== SyntaxKind.NullKeyword) {
+        if (t === SyntaxKind.LooseEqual) {
+          parser.onError(
+            DiagnosticSource.Parser,
+            DiagnosticKind.Lint,
+            diagnosticMap[DiagnosticCode._may_be_unsafe_if_you_are_relying_on_type_coercion],
+            parser.curPos,
+            parser.pos
+          );
+          if (context & Context.Autofix) {
+            parser.token = SyntaxKind.StrictEqual;
+          }
+        }
+        if (t === SyntaxKind.LooseNotEqual) {
           parser.onError(
             DiagnosticSource.Parser,
             DiagnosticKind.Lint,
-            diagnosticMap[DiagnosticCode.Forbidden_bitwise_operation],
+            diagnosticMap[DiagnosticCode.Use_of_may_be_unsafe_if_you_are_relying_on_type_coercion],
             parser.curPos,
             parser.pos
           );
+          if (context & Context.Autofix) {
+            parser.token = SyntaxKind.StrictNotEqual;
+          }
+        }
       }
-    }
 
+      if (parser.subRules && SubRules.Bitwise) {
+        switch (t) {
+          case SyntaxKind.BitwiseAnd:
+          case SyntaxKind.BitwiseAndAssign:
```

Comment:
@aladdin-add No worries. This switch case is huge and ugly. I will soon enough pack it down in some bitwise masks for perf reasons